### ISSUE TITLE
.as misidentified as AngelScript

### ISSUE THREAD

**Issue by meisels** [github/linguist#5052]
I uploaded ActionScript files to my repository and they are defined as AngelScript.

## Problem Description

If you search for the extension .as, the results will be mostly ActionScript. I've made an override but now whenever you click on the ActionScript text (in the language panel), it won't bring up my .as files because they are under the tag AngelScript still.

### URL of the affected repository:
https://github.com/meisels/gamething

### Last modified on:
2020-10-15

### Expected language:
ActionScript

### Detected language:
AngelScript

**Comment by Alhadis** [comment]
This should be fixed by #4926.

**Comment by meisels** [comment]
The problem is I'm using ActionScript 2 and not 3. That may be why it's not being detected.

I overrode linguist, as you can see below:
![image](https://user-images.githubusercontent.com/68653653/96250239-86641780-0f7c-11eb-979a-3fbae5eb9e3a.png)

But when you search through the code, .as files are still identified as AngelScript:
![image](https://user-images.githubusercontent.com/68653653/96250397-c6c39580-0f7c-11eb-9050-6238f1772fe9.png)

Why is it still classifying the files as AngelScript there when I manually corrected them here (in .gitattributes):
![image](https://user-images.githubusercontent.com/68653653/96250560-04c0b980-0f7d-11eb-9b58-cf923dc252db.png)

**Comment by lildude** [comment]
> Why is it still classifying the files as AngelScript there when I manually corrected them here (in .gitattributes):

Because modifying _just_ the `.gitattributes` isn't enough to force a re-analysis (by Linguist) and re-indexing by the search engine (responsible for your second screenshot) and you've not made any other changes in the last 9-10 hours. Changing another file should force a re-analysis and re-indexing, though it can take a bit of time for both to happen, especially during US working hours when GitHub.com is much busier.

**Comment by meisels** [comment]
Ah. That would make sense. Okay, the last time I edited it was about 6 hours ago so in about 4 hours I'll check.

**Comment by lildude** [comment]
You need to modify another file, _not_ the `.gitattributes`, to trigger a re-analysis. If in doubt, modify one of your ActionScript files that show up in the search results.

**Comment by meisels** [comment]
It's been 4 hours and nothing's changed. Is there any way I could override everything and just set the languages myself? Or remove the languages tab entirely?

**Comment by meisels** [comment]
I just checked and the files are still labeled as AngelScript.

**Comment by lildude** [comment]
🤔 I'm really not sure then. You'll need to contact GitHub support then as this is something with the search indexing which is independent of Linguist. Things are likely to improve when a new release is made and deployed which pulls in the changes from https://github.com/github/linguist/pull/4926. I'm not sure when that will be though, possibly not until the end of October, early November.

**Comment by meisels** [comment]
@lildude I see you forked my repository and added a line of code. What does // Foo do? I'm changing it now to see if it works.
![image](https://user-images.githubusercontent.com/68653653/96375761-ec04ff00-1148-11eb-8b12-5f6a270118d0.png)

**Comment by lildude** [comment]
It doesn't do anything. It was an attempt to force re-indexing, but it didn't work 😄

**Comment by meisels** [comment]
:sob: So there's no way to just remove that feature entirely?

**Comment by lildude** [comment]
> 😭 So there's no way to just remove that feature entirely?

Not really... it's the main search functionality behind the site and would only affect those that choose to search just your repo for files with the ActionScript language. I suspect this will be a very very very small number of people. As I mentioned before, things may improve once the recent updates have been deployed.

**Comment by CodeAsherPlayz** [comment]
Hmmm, it is 2 December when I commented on this. Has this issue been rectified in #5089? If not, please look into it and add the Misinterpreted Language tag. Thanks!

**Comment by lildude** [comment]
> Has this issue been rectified in #5089?

Yes, your original issue of the language being incorrectly identified _has_ been resolved by #5089:

```console
$ git clone -q git@github.com:meisels/gamething.git
$ cd gamething/
$ git rm .gitattributes
rm '.gitattributes'
$ git commit -m 'Remove .gitattributes'
[master 3575377] Remove .gitattributes
 1 file changed, 16 deletions(-)
 delete mode 100644 .gitattributes
$ github-linguist . --breakdown
57.29%  JavaScript
23.11%  ActionScript
11.66%  HTML
7.04%   NSIS
0.90%   CSS

ActionScript:
code/code.as
code/functions.as

HTML:
index.html
trophyapi/Auth.html
trophyapi/Unlock.html

NSIS:
register/Register Game Thing.nsi

CSS:
style.css

JavaScript:
trophyapi/MD5.js

$
$ git remote add lildude git@github.com:lildude/effective-octo-fortnight.git
$ git push lildude --all
Enumerating objects: 326, done.
Counting objects: 100% (326/326), done.
Delta compression using up to 4 threads
Compressing objects: 100% (172/172), done.
Writing objects: 100% (326/326), 4.30 MiB | 996.00 KiB/s, done.
Total 326 (delta 134), reused 324 (delta 133), pack-reused 0
remote: Resolving deltas: 100% (134/134), done.
To github.com:lildude/effective-octo-fortnight.git
 * [new branch]      master -> master
$
```

... and you can see the re-analysis done by GitHub reflects this in the repo I've just pushed your code to: https://github.com/lildude/effective-octo-fortnight

![Screenshot 2020-12-02 at 10 19 50](https://user-images.githubusercontent.com/627280/100860069-eb6dc100-3487-11eb-9a5d-f0392ac6c769.png)

What hasn't changed or been fixed is your subsequent issue with your search results not being correct.  As I mentioned before, the search functionality is independent of Linguist and as such would be an issue with the search side of things and as such I advised you to contact GitHub support.

I note you've also not changed either of your ActionScript files since https://github.com/github/linguist/pull/5089 was deployed so your repo will not have been reanalysed nor reindexed since October so will still be using old cached data.

With that in mind, I recommend you try changing one or both of your ActionScript files and give it time for the search to reindex your repo. If it still continues to report AngelScript in the code search, contact GitHub support.

**Comment by lildude** [comment]
The code search behaviour is not due to Linguist and Linguist has no influence over it. The search issue is detailed at https://github.com/github/linguist/discussions/5249#discussioncomment-428558
